CHANGELOG.md

```diff
@@ -6,6 +6,9 @@ Please follow the format in [Keep a Changelog](http://keepachangelog.com/)
 
 ## [Unreleased]
 
+- Drop Ruby 3.1 support.  Add >= 3.2 ruby support in gemspec.  EOL for 3.1.0 was 2025-03-31
```

Comment:
This change and the ones adding `bigdecimal` are related to the Ruby 3.1x removal, right? If so, how about removing them from this PR to clean the diff?

Reply:
Bigdecimal was in prep for ruby removing bigdecimal in the core language as of 3.4 - https://rubyreferences.github.io/rubychanges/3.4.html#default-gems-that-became-bundled

Could remove those from the gemfiles but each additional version at 3.4 and after will need them